xbmc/addons/PVRClient.cpp

```diff
@@ -227,9 +229,29 @@ PVR_CONNECTION_STATE CPVRClient::GetConnectionState(void) const
 void CPVRClient::SetConnectionState(PVR_CONNECTION_STATE state)
 {
   CSingleLock lock(m_critSection);
+
+  if (m_connectionState == PVR_CONNECTION_STATE_CONNECTED)
+    m_ignoreClient = false;
+  else if (m_connectionState == PVR_CONNECTION_STATE_CONNECTING &&
+           m_prevConnectionState == PVR_CONNECTION_STATE_UNKNOWN)
+    m_ignoreClient = true;
+
+  m_prevConnectionState = m_connectionState;
```

Comment:
Not sure what you're trying to achieve here. Why can't you just compare against "state"?